src/lib/validate.c

```diff
@@ -887,6 +887,7 @@ pgp_validate_file(pgp_io_t *             io,
             }
         }
     }
+    (void) close(outfd);
```

Comment:
I think it should be ``if (!strcmp(outifle, "-")) (void) close(outfd);`` otherwise you may endup closing STDOUT_FILENO (see above)